core/src/main/scala/com/avsystem/commons/SharedExtensions.scala

```diff
@@ -463,9 +463,13 @@ object SharedExtensionsUtils extends SharedExtensions {
       if (tr.isFailure) OptArg.Empty else OptArg(tr.get)
 
     /**
-      * Apply side-effect only if Try is a failure.
+      * Apply side-effect only if Try is a failure. The provided `action` function will be called with the
+      * throwable from the failure case, allowing you to perform operations like logging or error handling.
+      * 
+      * Any exceptions thrown by the `action` function are caught and ignored, ensuring that this method
```

Comment:
This is not true. You've literally just added a test that shows it's not all exceptions.